contracts/0.4.24/oracle/OracleChiCaller.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: 2020 Lido <info@lido.fi>
+
+// SPDX-License-Identifier: GPL-3.0
+
+/* See contracts/COMPILERS.md */
+pragma solidity 0.4.24;
+
+import "../interfaces/ILidoOracle.sol";
+
+interface IFreeFromUpTo {
+    function freeFromUpTo(address from, uint256 value) external returns (uint256 freed);
+    function balanceOf(address account) external view returns (uint256);
+    function approve(address spender, uint256 amount) external returns (bool);
+}
+
+
+contract OracleChiCaller {
+    IFreeFromUpTo public constant chi = IFreeFromUpTo(0x0000000000004946c0e9F43F4Dee607b0eF1fA1c);
+    ILidoOracle public lidoOracle;
+
+    constructor(ILidoOracle _oracle) public {
+        lidoOracle = _oracle;
+    }
+
+    function reportBeacon(uint256 _epochId, uint128 _beaconBalance, uint128 _beaconValidators) external {
+        uint256 gasStart = gasleft();
```

Comment:
can we move all gas calculations offchain. we're gas golfing, so we can save a bit on gas by counting the amount of token needed offchain and just pass a number of tokens to burn here

Reply:
looks no, if we want more or less accurate calculations. But yes, if we just estimate average gas consumption and do not to be too accurate. Do you mean this?